rfcs/0008-commitstatus/README.md

```diff
@@ -0,0 +1,63 @@
+
+# RFC-xxxx Create a new API kind for Commit Status
+
+**Status:** provisional
+
+**Creation date:** 2022-05-12
+
+**Last update:** 2022-05-12
+
+## Summary
+
+There should be a dedicated kind in the notification controller for sending commit status notifications to git providers.
+
+## Motivation
+
+Currently, The Alert type can reference both git providers and chat providers. However, the differences between the two providers and how notifications being sent to them should be handled has continued to diverge. 
+For example, there is a limit on the status message for git provider and sending the same notification twice overwrites the first one.
+
+By creating a sepeerate kind for git commit status, the specific nuances of the provider can be properly taken into account.
+
+### Goals
+
+- Add a new kind `CommitStatus` to Flux notification controller for handling commit status notifications.
+
+### Non-Goals
+
+- Add new `spec` fields to the `Alert` kind.
+
+## Proposal
+
+Introduce a new kind in the notification-controller called `CommitStatus`.
+The `CommitStatus` kind will similar to the `Alert` kind, referencing a provider and including event sources to accept events from. The major difference is that it will only reference git providers. Additionally, it can only have a `Kustomization` as its event source since it requires the `revision` field in the event metadata.
+
+This proposal will also introduce breaking changes as the `Alert` kind will no longer send notifications to git providers.
+
+### User Stories
+
+If a user want to receive status notification on github for a deployment, a `CommitStatus` kind can be used:
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1beta1
+kind: CommitStatus
+metadata:
+  name: webapp
+  namespace: flux-system
+spec:
+  providerRef: 
+    name: github # The `spec.type` of the provider has to be github, gitlab, or bitbucket
+  eventSeverity: info
+  eventSources:
+    - kind: Kustomization
+      name: webapp
```

Comment:
We should probably have a `summary` field here similar to [`Alert`](https://fluxcd.io/docs/components/notification/alert/#specification). This can be helpful in adding some metadata about the event, which can be helpful, for example when there are multiple clusters. (ref: https://github.com/fluxcd/notification-controller/issues/194)

Reply:
The name `summary` doesn't make sense for commit status IMO, it should be `prefix` or something else.

Reply:
We could use `context`. 
`context` is the name of the field the GitHub and GitLab API . They both describe it as
```
A string label to differentiate this status from the status of other systems.
```